include/libhal-arm-mcu/stm32f1/adc.hpp

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Khalil Estell
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LIC`ENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+
+#include <libhal/adc.hpp>
+#include <libhal/initializers.hpp>
+#include <libhal/units.hpp>
+
+namespace hal::stm32f1 {
+/**
+ * @brief Analog to digital converter
+ *
+ */
+class adc final : public hal::adc
+{
+public:
+  /// Channel specific information
+  struct channel
+  {
+    /// ADC port
+    std::uint8_t port;
+    /// ADC pin
+    std::uint8_t pin;
+    /// Channel data index
+    uint8_t index;
```

Comment:
Since libhal 4.5.0 you can now use `hal::u8`, `hal::i8` and other `hal::uX` types. Its a shorter version of the standard integer sizes based on Rust and Zig. Lets switch over to those.